File: Ad-Hoc/__3000__/3489/3489.cpp
#include <iostream>
#include <vector>
#include <sstream>
#include <iomanip>
#include <cmath> // fabs para comparação com double

using namespace std;

class CppTemplatesAndSTLContainers {
public:
    static void findIndicesWithSum(const vector<double>& array, double target) {
        bool has_value = false;
        const double EPS = 1e-9; // tolerância para comparação de double

        for (size_t i = 0; i < array.size(); i++) {
            for (size_t j = i + 1; j < array.size(); j++) {
                if (fabs(array[i] + array[j] - target) < EPS) {
                    cout << i << " " << j << " "
                              << fixed << setprecision(0) << target << "\n";
                    has_value = true;
                }
            }
        }
        if (!has_value) {cout << "null value\n";}
    }
};

int main() {
    vector<double> arr;
    double target, value;
    string line;

    // lê a linha inteira de números
    getline(cin, line);
    // lê o target em seguida
    cin >> target;

    // transforma a linha em números
    stringstream ss(line);
    while (ss >> value) {
        arr.push_back(value);
    }
    
    CppTemplatesAndSTLContainers::findIndicesWithSum(arr, target);
    return 0;
}

File: Ad-Hoc/__3000__/3489/3489.py
class Cpp_Templates_and_STL_Containers:
    @staticmethod #-> O(n²)
    def findIndicesWithSum(vector, target):
        has_result = False
        for i in range(len(vector)):
            for j in range(i + 1, len(vector)):
                if abs((vector[i] + vector[j]) - target) < 1e-9:  # tolerância para float
                    print(f"{i} {j} {target:.0f}")
                    has_result = True
        
        if not has_result: print("null value")
    
vector, target = [float(x) for x in input().split()], float(input())
Cpp_Templates_and_STL_Containers.findIndicesWithSum(vector, target)